racksdb/web/ui/src/components/ComboBox.vue

```diff
@@ -0,0 +1,123 @@
+<!--Copyright (c) 2022-2023 Rackslab
+
+This file is part of RacksDB.
+
+SPDX-License-Identifier: GPL-3.0-or-later -->
+
+<script setup lang="ts">
+import { ref, computed, watchEffect, onMounted } from 'vue'
+import type { Ref } from 'vue'
+import { useRouter } from 'vue-router'
+import type { Datacenter, Infrastructure } from '@/composables/RacksDBAPI'
+import {
+  Combobox,
+  ComboboxButton,
+  ComboboxInput,
+  ComboboxOption,
+  ComboboxOptions
+} from '@headlessui/vue'
+import { ChevronUpDownIcon, CheckIcon } from '@heroicons/vue/24/outline'
+
+const selectedItem: Ref<string | undefined> = ref()
+const router = useRouter()
+const input = ref('')
+const inputComponent: Ref<typeof ComboboxInput | null> = ref(null)
+const filteredItem = computed(() =>
+  input.value === ''
+    ? props.items
+    : props.items.filter((item) => {
+        return item.name.toLocaleLowerCase().includes(input.value.toLocaleLowerCase())
+      })
+)
+type itemType = Datacenter | Infrastructure
+
+function goToItem(item: string) {
+  router.push({
+    name: props.item + 'details',
+    params: { name: item }
+  })
+}
+
+const props = defineProps({
+  item: String,
```

Comment:
The name `item` might be too generic. This prop holds the name of the type of items manipulated by the combobox used to define the targeted route name eventually. A better name could be `itemType`. It would be great to restrict its type to a union of literals such as `"datacenter" | "infrastructure"`, thus avoiding any free form string content.